### PR TITLE
SP026. User update

### PR DESCRIPTION
**Pull request by HumRuslan**
**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI

### CODE REVIEW

**Comment by HumRuslan** [comment]
change_password
profile_page

File: app/controllers/account/user/profile_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  before_action :find_user
+
+  def edit; end
+
+  def update
+    if params[:user].key?(:current_password)
+      flash[:notice] = "Profile is saved" if @user.update_with_password(user_params)
+    elsif @user.update(user_params)
+      flash[:notice] = "Profile is saved"
+    end
```
Comment (chu4) [inline app/controllers/account/user/profile_controller.rb:11]: a валідаційні еррори виводити?

File: app/controllers/account/user/profile_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class Account::User::ProfileController < Account::User::UserApplicationController
+  before_action :find_user
+
+  def edit; end
+
+  def update
+    if params[:user].key?(:current_password)
+      flash[:notice] = "Profile is saved" if @user.update_with_password(user_params)
+    elsif @user.update(user_params)
```
Comment (chu4) [inline app/controllers/account/user/profile_controller.rb:9]: update_with_password не працює для обох випадків?
по документації мало б

File: app/controllers/account/user/profile_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Account::User::ProfileController < Account::User::UserApplicationController
+  before_action :find_user
+
+  def edit; end
+
+  def update
+    if params[:user].key?(:current_password)
```
Comment (chu4) [inline app/controllers/account/user/profile_controller.rb:7]: `if params[:user].key?(:current_password) && @user.update_with_password(user_params)`

Comment (chu4) [inline app/controllers/account/user/profile_controller.rb:7]: `if user_param.key?(:current_password)`

File: app/uploaders/avatar_uploader.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  include CarrierWave::MiniMagick
+  process resize_to_fit: [400, 400]
+
+  # Override the directory where uploaded files will be stored.
+  # This is a sensible default for uploaders that are meant to be mounted:
+  def store_dir
+    "uploads/avatar/#{model.id}"
+  end
+
+  # Provide a default URL as a default if there hasn't been a file uploaded:
```
Comment (chu4) [inline app/uploaders/avatar_uploader.rb:13]: ці коменти можна вже зачищати, так як то вж ене перший твій аплоудер

**Review by chu4 — APPROVED** [review]

**Pull request merged**